docs/reference/cpp/index.md

```diff
@@ -0,0 +1,172 @@
+    pay close attention to tutorials concerning them!
+
+| Resource               | Description                                                                                 |
+| :--------------------- | :------------------------------------------------------------------------------------------ |
+| [w3schools Tutorial](https://www.w3schools.com/cpp/default.asp) | A structured tutorial that goes through basic concepts in C++. It's good to do up to the section on Classes. |
+| [YouTube Tutorial](https://youtu.be/vLnPwxZdW4Y) | If you prefer video tutorial, then this is a comprehensive 4 hour video covering similar concepts to the one above. It is 4 hours long though. |
+| [Dynamic Memory Overview](https://www.tutorialspoint.com/cplusplus/cpp_dynamic_memory.htm) | A page going over how dynamic memory works in C++. |
+
+Feel free to add other resources other than the ones listed above if you find any that you like!
+
+## Differences Between C and C Plus Plus
+
+For most use cases, you can think of C++ as a superset of C. While this is not technically true, more often than not
+you are able to write standard C code for a C++ program without issues. However, doing so ignores a lot of the benefits
+and reasons to use C++.
+
+### Classes and Structs
+
+In C structs can only contain member variables, but in C++ structs are basically classes but with a default member
+visibility of public instead of private.
+
+!!! example
+    The following code blocks are equivalent.
+
+    ```C++
+    struct foo {
+    private:
+        int x;
+        void helper(void);
+    public:
+        foo(int y);
+    }
+    ```
+
+    ```C++
+    class foo {
+    private:
+        int x;
+        void helper(void);
+    public:
+        foo(int y);
+    }
+    ```
+
+### Namespaces
+
+One problem that is prevalent in C concerns the scoping of names. For example, let there be two files `A.h` and `B.h`
+and a program `ighxy.c` wants to use both of them.
+
+```C
+// A.h
+float x;
+int bar(void);
+
+// B.h
+float x;
+int bar(void);
+
+// ighxy.c
+#include "A.h"
+#include "B.h"
+
+int main(void) {
+    int a = bar();
+    ...
+}
+/* Error, does not compile*/
+```
+
+Our program cannot compile because the linker cannot distinguish which `bar()` function we want to use! One way to fix
+this in a C program would be to rename them `a_bar()` and `b_bar()`. Although this fix seems trivial for this example,
+applying it to a file that has potentially 100 functions can be a lot more difficult, especially if two files just
+happen to share the same prefix for their functions!
+
+C++ introduces namespaces.
+
+```C++
+// A.h
+namespace a {
+float x;
+int bar(void);
+}  // namespace a
+
+// B.h
+namespace b {
+int bar(void);
+}  // namespace b
+
+// ighxy.cpp
+#include "A.h"
+#include "B.h"
+
+int main(void) {
+    int a = a::bar();
+    int b = b::bar();
+    float xa = a::x;
+    float xb = b::x;
+    ...
+}
+```
```

Comment:
I think this could be tabbed content: https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#grouping-code-blocks

You could also consider taking advantage of titled code blocks: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title